src/main/kotlin/org/rust/ide/inspections/RsNamingInspection.kt

```diff
@@ -100,6 +97,11 @@ open class RsSnakeCaseNamingInspection(elementType: String) : RsNamingInspection
         }
         return Pair(false, if (str.isEmpty()) "snake_case" else name.toSnakeCase(false))
     }
+
+    override fun isSuppressedFor(element: PsiElement): Boolean {
+        val function = element.parent as? RsFunction ?: return false
+        return function.findOuterAttr("no_mangle") != null
```

Comment:
Hm, looks like we should suppress annotation only if a function has `extern` modifier. Otherwise, the compiler still produces a warning

Reply:
You're right, I changed it.